### PR TITLE
Topic recognition: (s48336813) Classifying Alzheimer's disease (normal and AD) on the ADNI dataset using a GFNet vision transformer.

### PR DESCRIPTION
**Pull request by sbravo-puppet**
This pull request provides the implementation for classifying Alzheimer's disease (normal and AD) using the ADNI dataset, leveraging the GFNet vision transformer.

Key Contributions:

- Preprocessing and augmentation of the ADNI dataset.
- Development of a GFNet-based transformer model tailored for image classification (testing accuracy 70.82%).
- Model training with hyperparameter tuning, optimised for Alzheimer's disease classification.
- Evaluation of model performance with accuracy metrics, plots and visualisations.

This submission includes:

- modules.py: GFNet vision transformer architecture.
- dataset.py: Data handling and preprocessing.
- train.py: Training pipeline.
- predict.py: Performance evaluation.
- utils.py: Utility functions.
- README.md: Instructions for replicating the experiments and findings.
- resources/: Folder collection of images for README.

Sean Bourchier
s48336813

### CODE REVIEW

**Comment by aniketgupta17** [comment]
Observational Feedback

Pull Request:
Correctly created the Pull request from Topic Recognition Branch .
The pull request shows no incorporation of previous feedback .

File Organizing: Well-organized files.

Commit Log:
Commit messages are meaningful .
Commits demonstrate a logical development flow.
Commits are regularly made, showing incremental development rather than bulk updates.

Documentation:
The README file provides a clear explanation of the algorithm and the problem it solves.
Code comments and docstrings  are included .
Proper GitHub markdown formatting is used, with organized headings, lists, and code blocks.
Can add about Conclusion of the report .

**Comment by gayanku** [comment]
## Marking
<table><tbody>
  <tr><td colspan='3'><b>Good/OK/Fair Practice (Design/Commenting, TF/Torch Usage)</b></td> </tr>
	<tr><td></td><td>Good design and implementation. </td><td></td></tr>
	<tr><td></td><td>Spacing and comments. </td><td></td></tr>
	<tr><td></td><td>Header blocks. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Recognition Problem</b></td></tr>
	<tr><td></td><td>OK solution to problem. Some overfitting.</td><td>-1</td></tr>
	<tr><td></td><td>Driver Script present. </td><td></td></tr>
	<tr><td></td><td>File structure present. </td><td></td></tr>
	<tr><td></td><td>Good Usage & Demo & Visualisation & Data usage. </td><td></td></tr>
	<tr><td></td><td>Module present. </td><td></td></tr>
	<tr><td></td><td>Commenting present. </td><td></td></tr>
	<tr><td></td><td>No Data leakage found. Uses Data Aug</td><td></td></tr>
	<tr><td></td><td>Difficulty : Hard.  GFNet (Hard Difficulty)</td><td></td></tr>
	
  <tr><td colspan='3'><b>Commit Log</b></td></tr>
	<tr><td></td><td>Good Meaningful commit messages. </td><td></td></tr>
	<tr><td></td><td>Good Progressive commits. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Documentation</b></td></tr>
	<tr><td></td><td>Readme :Good. </td><td></td></tr>
	<tr><td></td><td>Model/technical explanation :Good. </td><td></td></tr>
	<tr><td></td><td>Description and Comments :Good. </td><td></td></tr>
	<tr><td></td><td>Markdown used and PDF submitted. PDF Checked.</td><td></td></tr>
	
  <tr><td colspan='3'><b>Pull Request</b></td></tr>
	<tr><td></td><td>Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch). </td><td></td></tr>
	<tr><td></td><td>No Feedback required. </td><td></td></tr>
	<tr><td></td><td>Request Description is good. </td><td></td></tr>
  <tr><td colspan='2'><b>TOTAL</b></td><td>-1</td></tr>
</tbody></table>

Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness.
_Subject to approval from Shakes_

**Comment by shakes76** [comment]
Good work
